### PR TITLE
Adding helper methods for interfaces

### PR DESCRIPTION
**Pull request by stefantalen**
While working on a Rector for rector-symfony I stumbled upon some repeated logic across multiple rules.

I found some classes that might be suitable for these functions so other Rector libraries can benefit from them (Since they need rector-src). Only issue is PHPStan notices they're not being used in this project.

Any thoughts?

### CODE REVIEW

**Comment by samsonasik** [comment]
You can add `@api` docblock above method so it will be skipped from unused method check

**Comment by TomasVotruba** [comment]
Hi, actually the other direction is preffered. The core should contain only core logic, and the framework-package should handle lower logic themselves.

Before, we had all "helper" methods in here, but they were often used only in 1 framework-package. 
The core what bloated and contained dozens of unused function. As you see, it also breaks the dead-method detection and could cause false positives we'll have to maintain.

In short, these PRs should be avoided and instead `@api` removed and public elements moved directly to their vendor packages.

**Review by TomasVotruba — CHANGES_REQUESTED** [review]
These PRs should be avoided.

**Comment by stefantalen** [comment]
I see where you're coming from.

To me it feels odd to have basic PHP checks like these put away in framework specific packages, but I'll add them to rector-symfony

**Comment by TomasVotruba** [comment]
The reason we had 100 helper method for any AST change. The abstraction was so extreeme we had dozens of unused methods we had to maintain.
Sometimes the good old AST with 5 lines is much more clearer and easy to use right in the spot :)
